medusa/name_parser/parser.py

```diff
@@ -461,6 +466,16 @@ def get(self, name):
             logger.debug("Using cached parse result for '{name}'", name=name)
             return self.cache[name]
 
+    def remove(self, indexer, indexer_id):
+        """Remove cache item given indexer and indexer_id."""
+        if not indexer or not indexer_id:
+            return
+        to_remove = (cached_name for cached_name, cached_parsed_result in self.cache.items() if
```

Comment:
It's still looping through all entries. Probably the data structure of the cache should be different where you could access all items that relates to a specific indexer / indexer id